### DISCUSSION TITLE
Wazuh Reporting Solutions to Vulnerbailities

### DISCUSSION THREAD

**Discussion by valkiriaaquatica** [wazuh/wazuh#25306] [Ideas]
Hi!!, 
Right now, in Wazuh and Wazuh agent when they collect information about machinesand with it theirvulnerabilities and then exposes these vulns and gives information about each one, so far so good.
There are other vulnerability agents on the market such as Nessus that does a similar function, but when it exposes this data it almost always offers a “**Solution**” section. 
For example, if Tenable were to detect that an Ubuntu machine has this vulnerability: https://www.tenable.com/plugins/nessus/203144 would indicate a solution like this: “Update the affected packages.”  which is very useful as a guideline.

I miss this in Wazuh, it is true that for example here https://www.reddit.com/r/Wazuh/comments/1egmig4/does_wazuh_report_solutions_to_vulnerabilities/ was talked about it and it was commented that there are some cases where “condition” would give you an idea (there are rare vulnerabilities that have it).

However, if for example from my point of view, you want to do automatic patching of computers based on real time information, either a simple patching like “Update package” or more complex solutions. 

This would be a correct example of a package in this case from “Git” which could offer a solution which is below in the json: 
 - “solution”: “Update the repository and then upate the packeg version”,

```
{
  "_index": "wazuh-states-vulnerabilities-wazuh.manager",
  "_id": "001_093bc19129f34ff3969df1b8097e6b8584e4dfa6_CVE-2018-1000021",
  "_score": 0,
  "_source": {
    "agent": {
      "id": "001",
      "name": "testing",
      "type": "wazuh",
      "version": "v4.8.1"
    },
    "host": {
      "os": {
        "full": "Ubuntu 24.04 LTS (Noble Numbat)",
        "kernel": "6.8.0-39-generic",
        "name": "Ubuntu",
        "platform": "ubuntu",
        "type": "ubuntu",
        "version": "24.04"
      }
    },
    "package": {
      "architecture": "amd64",
      "description": "fast, scalable, distributed revision control system",
      "name": "git",
      "size": 21692,
      "type": "deb",
      "version": "1:2.43.0-1ubuntu7.1"
    },
    "vulnerability": {
      "category": "Packages",
      "classification": "CVSS",
      "description": "GIT version 2.15.1 and earlier contains a Input Validation Error vulnerability in Client that can result in problems including messing up terminal configuration to RCE. This attack appear to be exploitable via The user must interact with a malicious git server, (or have their traffic modified in a MITM attack).",
      "detected_at": "2024-08-06T21:23:08.254Z",
      "solution": "Update the repository and then upate the package version",
      "enumeration": "CVE",
      "id": "CVE-2018-1000021",
      "published_at": "2018-02-09T23:29:00Z",
      "reference": "http://www.batterystapl.es/2018/01/security-implications-of-ansi-escape.html",
      "scanner": {
        "vendor": "Wazuh"
      },
      "score": {
        "base": 6.8,
        "version": "2.0"
      },
      "severity": "Medium"
    },
    "wazuh": {
      "cluster": {
        "name": "wazuh.manager"
      },
      "schema": {
        "version": "1.0.0"
      }
    }
  },
  }
}

```

As a summary, for automations done with for ex Ansible, I think it would be a good idea that Wazuh give recommendations on how to fix vulnerabilities in the Vulnerability Dteection -> Inventory section so we can fix them all in one at the same time, considering the case I ahve 50 servers, and two main common vulns in 40 of them, so I applied 2 playbooks that grasp data from Wazuh Api and then apply the patch to the servers automatically (in this case for ex with Ansible).

**Comment by Damian-Mangold** [comment]
Hi @valkiriaaquatica, thank you for your interesting proposal; we agree that it would be an excellent enhancement to the vulnerability detector. However, it’s important to note that the content currently provided by vendors does not consistently include information on the actions required to resolve a vulnerability. 
We will investigate further to determine whether this information can be obtained from an alternative source, and if feasible, we will consider incorporating this new content.

Thank you very much for the suggestion.

**Reply by valkiriaaquatica** [reply]
Nice to hear that @Damian-Mangold , would be a huge improvement!
